### DISCUSSION TITLE
Activate AMR only after criterion?

### DISCUSSION THREAD

**Discussion by mangerij** [idaholab/moose#17125] [Q&A Modules: Phase field] [answered]
Is it easy to set in the input file to activate AMR _only_ after percent change in energy, F, is below a certain value? Say: if dF/dt = 1e-3 start to aggressively coarsen? 

I also seem to have problems getting the simulation to immediately refine down to max_h_level and stay that way. Say for example I use:

initial_steps = 5

but I want ALL elements (and not just markers/indicators) to be refined down to this until dF/dt = 1e-3 when the markers/indicators kick in. Then it should coarsen the domains (image attached).

Things are looking peachy here in ferroelectrics land (big thanks to @SudiptaBiswas for helping us get the "thin film" boundary conditions working). 

cheers
John

![WhatsApp Image 2021-02-23 at 10 01 15 AM(1)](https://user-images.githubusercontent.com/7597716/109061041-6a207600-76e6-11eb-9492-37601621ab2a.jpeg)
![WhatsApp Image 2021-02-23 at 10 01 15 AM](https://user-images.githubusercontent.com/7597716/109061042-6ab90c80-76e6-11eb-8723-d1f5db371fbf.jpeg)

**Comment by GiudGiud** [comment]
Hi John

I dont think AMR can be turned on/off using the Control system.

However you could use an AuxVariable to control the refinement criteria, and those could be set using a criterion like the one you mentioned. 

Guillaume

**Reply by aeslaughter** [reply]
It should be possible to create a control that does, but it might be just as easy to create a custom Marker object that can check the criteria. It could use or not-use the results from another marker.

**Comment by mangerij** [comment, marked as answer]
I am still thinking about this.

We have an order parameter that is a vector whose components start near zero and small. And randomly oriented so they possible have large gradients. We ideally want to refine down (so value threshold could work initially) but we don't want to refine too far at the early stages.

But later in the simulation the value of some of the components are much larger but some could be small still!

At this point we want to have maximum refinement near the phase field interface but maximum coarsening far away. It's pretty tricky. I will try to post a small example...

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Reply by aeslaughter** [reply]
Sounds like a case for a custom marker. I made one a long while ago to track a phase-field interface but account for the direction of travel and marked more in the direction of travel.

**Reply by mangerij** [reply]
This makes sense. As our domains (in the above image) merely coarsen until we kill the simulation. This is what I aim to tackle with the AMR.

In the above picture, the domain walls (interface between "phases") lie at inclined planes. An example, being [110] and all symmetry equivalent directions.

It would be nice to not have to assume this direction of the propagating wall for the Marker/Indicator since different materials have symmetry specific wall angles with respect to a global axis and these can change drastically with temperature.
